### PR TITLE
introduce a crypto::recover_secp_public_key syscall.

### PR DESCRIPTION
**Pull request by raulk**
Supersedes https://github.com/filecoin-project/ref-fvm/pull/627 with a clean diff, refresh from master, and some small changes (check commit log).

### CODE REVIEW

File: fvm/src/call_manager/default.rs [hunk 1]
```diff
@@ -351,6 +351,15 @@ where
...
 
+        // prepare actor code
+        // TODO remove once the EVM smart contract actor is integrated in the builtin-actors bundle
+        //  https://github.com/filecoin-project/ref-fvm/issues/693
+        self.engine()
+            .load_code_cid(&state.code, self.blockstore())
+            .map_err(
+                |_| syscall_error!(NotFound; "actor code cid does not exist {}", &state.code),
+            )?;
+
```
Comment (raulk) [inline fvm/src/call_manager/default.rs:362]: Tracked in #708.

Comment (Stebalien) [reply fvm/src/call_manager/default.rs:362]: Isn't this part of a separate patch?

- Introduce a secp recover syscall.
- Code loading?

File: fvm/src/machine/engine.rs [hunk 2]
```diff
@@ -208,6 +208,33 @@ struct Cache<K> {
 }
 
 impl Engine {
+    /// Attempts to load Actor Code CID from the blockstore and
+    /// instantiate a wasmtime Module from the WASM bytecode if the
+    /// given code CID is present, otherwise returns None.
+    ///
+    /// Implicitly caches instantiated modules.
+    pub fn load_code_cid<BS: Blockstore>(
```
Comment (Stebalien) [inline fvm/src/machine/engine.rs:216]: `install_code`? We don't actually need to return it.

**Review by Stebalien — APPROVED** [review]
Logic LGTM. I'd do this in two PRs as one is a new syscall, and one supports EVM code loading. But I don't feel too strongly about that.

I'd also change `load_code` so we don't have to clone the module on every call (we don't use the return value).

**Review by mriise — APPROVED** [review]
LGTM, +1 on splitting this PR though (not the end of the world if it isnt though).

File: fvm/src/machine/engine.rs (hunk 2 above)
Comment (raulk) [reply fvm/src/machine/engine.rs:216]: SGTM. I'll move this part to a different patch and we can address there.

**Comment by raulk** [comment]
The extra feature has been moved to #711 and this PR is now strictly about the recover_secp_public_key syscall.

**Pull request merged**